gprofiler/utils.py

```diff
@@ -117,6 +121,14 @@ def wait_event(timeout: float, stop_event: Event, condition: Callable[[], bool])
             raise TimeoutError()
 
 
+def wait_for_file(pattern: Union[str, Path], timeout: float, stop_event: Event) -> Path:
+    if isinstance(pattern, Path):
+        pattern = str(pattern)
+    files = wait_event(timeout, stop_event, lambda: glob.glob(pattern))
```

Comment:
You should not do the `glob` with no delay, that's a ton of IO calls

Reply:
`os.path.exists` should be more efficient, since you know the exact file path for some functions

Reply:
First, there is a delay in `wait_event` of 0.1s (via `Event.wait`). Second, `glob` actually does `os.path.exists` when the pattern does not contain any wildcards (`Lib/glob.py`):
```
...
    if not has_magic(pathname):
        if basename:
            if os.path.lexists(pathname):
                yield pathname
...
```

Reply:
Since there's a sleep, it doesn't matter at all. But if there wasn't any sleep, it wouldn't matter if we use `glob.glob` or `os.path.exists` because both would reach 100% CPU. It doesn't matter if you do 1 IO call or 20 IO calls if you do them repeatedly (there's shouldn't be any disk access here, it's just the CPU we're using)